node/src/test_utils/mod.rs

```diff
@@ -569,6 +569,44 @@ pub mod unshared_test_utils {
         pub assertions: Box<dyn FnOnce(&mut A) + Send>,
     }
 
+    pub fn capture_numbers_with_separators_from_str(
+        surveyed_str: &str,
+        digit_grouped_by: usize,
+        separator: char,
+    ) -> Vec<String> {
+        let regex =
+            format!("(\\d{{1,{digit_grouped_by}}}(?:{separator}\\d{{{digit_grouped_by}}})+)");
+        let re = regex::Regex::new(&regex).unwrap();
+        let captures = re.captures_iter(surveyed_str);
+        captures.map(|capture| capture[1].to_string()).collect()
+    }
+
+    #[macro_export]
+    macro_rules! assert_on_testing_enum_with_all_its_variants {
```

Comment:
I appreciate your effort, but I feel that introducing a macro for tests might be a bit far-fetched. It doesn't seem generic enough and offers limited utility. I would suggest considering porting the code directly into the tests instead. Thank you!

Reply:
I deleted it but I don't agree on the premise that tests should never use macros. :-)